tls/s2n_security_policies.c

```diff
@@ -980,6 +980,7 @@ const struct s2n_security_policy security_policy_rfc9151 = {
     .kem_preferences = &kem_preferences_null,
     .signature_preferences = &s2n_signature_preferences_rfc9151,
     .certificate_signature_preferences = &s2n_certificate_signature_preferences_rfc9151,
+    .certificate_preferences_apply_locally = true,
```

Comment:
I would vote yes. If a customer is using the "rfc9151" policy, the most likely guess we have is that they want to be compliant with rfc9151, in which case I think implementing a more correct behavior is a good thing. 

While some customers might have just randomly picked it, 
1. it's a named policy, which doesn't have any promise of stability
2. we shouldn't cause more pain to the people trying to achieve rfc9151 compliance just because someone else might not want that.